pkg/dockercreds/docker_creds.go

```diff
@@ -47,15 +53,41 @@ func (c DockerCreds) AppendToDockerConfig(path string) error {
 	return ioutil.WriteFile(path, configJsonBytes, 0600)
 }
 
-func (c DockerCreds) append(a DockerCreds) DockerCreds {
+func (c DockerCreds) append(a DockerCreds) (DockerCreds, error) {
 	if c == nil {
-		return a
+		return a, nil
+	} else if a == nil {
+		return c, nil
 	}
 
 	for k, v := range a {
-		c[k] = v
+		if contains, err := c.contains(k); err != nil {
+			return nil, err
+		} else if !contains {
+			c[k] = v
+		}
+	}
+
+	return c, nil
+}
+
+func (c DockerCreds) contains(reg string) (bool, error) {
+	if !strings.HasPrefix(reg, "http://") && !strings.HasPrefix(reg, "https://") {
+		reg = "//" + reg
+	}
+
+	u, err := url.Parse(reg)
+	if err != nil {
+		return false, err
 	}
-	return c
+
+	for existingRegistry := range c {
+		if (RegistryMatcher{}.Match(u.Host, existingRegistry)) {
```

Comment:
Can we update this method to new require a receiver and just be a public method in the pkg?

Reply:
Fixed 🙂